eox_nelp/course_experience/api/v1/serializers.py

```diff
@@ -0,0 +1,131 @@
+"""Serializers used for the experience views."""
+from django.shortcuts import get_object_or_404
+from rest_framework_json_api import serializers
+
+from eox_nelp.course_experience.models import LikeDislikeCourse, LikeDislikeUnit, ReportCourse, ReportUnit
+from eox_nelp.edxapp_wrapper.course_overviews import CourseOverview
+
+from .validators import validate_course_key, validate_usage_key
+
+
+class UnitExperienceSerializer(serializers.ModelSerializer):
+    """Class to configure serializer fot UnitExperience.
+
+    Args:
+        serializer (serializers.ModelSerializer): the model serializer from json api
+
+    """
+    username = serializers.CharField(
+        source="author.username", required=False, allow_blank=True
+    )
+
+    def validate_item_id(self, value):
+        """Validates the item_id.
+
+        Args:
+            value: The item_id to be validated.
+
+        Returns:
+            The item_id if it is valid.
+
+        Raises:
+            InvalidUsageKeyError: If the item_id is not valid.
+
+        """
+        validate_usage_key(value)
+        return value
+
+
+class LikeDislikeUnitExperienceSerializer(UnitExperienceSerializer):
+    """Class to configure serializer fot LikeDislikeUnitExperience.
+
+    Args:
+        UnitExperienceSerializer: the serializer for unit experiences.
+    """
+    class Meta:
+        """ Class to configure serializer with  model LikeDislikeUnit """
+
+        model = LikeDislikeUnit
+        fields = "__all__"
+
+
+class ReportUnitExperienceSerializer(UnitExperienceSerializer):
+    """Class to configure serializer fot ReportUnitExperience.
+
+    Args:
+        UnitExperienceSerializer: the serializer for unit experiences.
+    """
+    class Meta:
+        """ Class to configure serializer with  model ReportUnit """
+        model = ReportUnit
+        fields = "__all__"
+
+
+class CourseExperienceSerializer(serializers.ModelSerializer):
+    """Class to configure serializer for CourseExperience.
+
+    Args:
+        serializer (serializers.ModelSerializer): the model serializer from json api
+
+    """
+    username = serializers.CharField(
+        source="author.username", required=False, allow_blank=True
+    )
+    course_id = serializers.CharField(required=False)
+
+    def validate_course_id(self, value):
+        """Validates a course key.
+
+        Args:
+            value (str): The course key to validate.
+
+        Returns:
+            str: The course key if it is valid.
+
+        Raises:
+            ValidationError: If the course key is not valid.
+
+        """
+        validate_course_key(value)
+        return value
+
+    def create(self, validated_data):
+        """    Create a new CourseOverview object.
+
+        Args:
+            self: The CourseOverview object.
```

Comment:
self makes reference to the `CourseExperienceSerializer` instance,and I think is not common to include that but I'm not sure